cli/tsc/99_main_compiler.js

```diff
@@ -698,6 +698,10 @@ delete Object.prototype.__proto__;
           /** @type {Record<string, any[]>} */
           const diagnosticMap = {};
           for (const specifier of request.specifiers) {
+            if (core.opSync("op_is_cancelled", {})) {
```

Comment:
How expensive is this call? It might be good to throttle this and only check every X milliseconds (that seems to be what TypeScript does in tsserver). Let me know if you think I should implement that as part of this PR.